src/cone_detector.cpp

```diff
@@ -111,19 +126,156 @@ bool ConeDetector::annotateCb(victoria_perception::AnnotateDetectorImage::Reques
     return true;
 }
 
+void ConeDetector::kmeansImage(cv::Mat image) {
+    int height = image.rows;
+    int width = image.cols;
+    cv::Mat labels; // Maps each pixel in image to the result cluster index.
+    static const int cluster_count = 10; // This is arbitrary. I increased it until I got the results I wanted.
+    cv::TermCriteria criteria {cv::TermCriteria::COUNT, 100, 1};  // int type: count=> max iterations, int max count, double epsilon
+    cv::Mat centers;
+
+    // Convert data into shape required for KMEANS.
+    cv::Mat reshaped_image = image.reshape(1, image.cols * image.rows);
+    assert(reshaped_image.type() == CV_8UC1);
+    cv::Mat reshaped_image32f;
+    reshaped_image.convertTo(reshaped_image32f, CV_32FC1, 1.0 / 255.0);
+    assert(reshaped_image32f.type() == CV_32FC1);
+
+    // Do KMEANS analysis.
+    kmeans(reshaped_image32f,           // Floating-point matrix of input samples, one row per sample.
```

Comment:
There are no "using namespace" anywhere. I'm not sure what opened this up. I tried to fix all required namespace qualifiers. I'll fix others as we discover them.